util.go

```diff
@@ -48,11 +51,9 @@ func decode(buf []byte, out interface{}) error {
 func encode(msgType messageType, in interface{}, msgpackUseNewTimeFormat bool) (*bytes.Buffer, error) {
 	buf := bytes.NewBuffer(nil)
 	buf.WriteByte(uint8(msgType))
-	hd := codec.MsgpackHandle{
-		BasicHandle: codec.BasicHandle{
-			TimeNotBuiltin: !msgpackUseNewTimeFormat,
-		},
-	}
+	hd := codec.MsgpackHandle{}
+	hd.TimeNotBuiltin = !msgpackUseNewTimeFormat
+
```

Comment:
Callout: No change in the wire format of hd, just removing a direct reference to `code.BasicHandle`. Similar to this [change](https://github.com/hashicorp/raft/pull/647/files/eec38fd04e67bba4484cbccdbdfd4cd4e5f595f7#r2153914642).